### PR TITLE
generate and maintain token ids

### LINKED ISSUE #46: Maintain tokenids on the server, doesn't work correctly right now

### PR DESCRIPTION
**Pull request by angelagilhotra**
closes #46

### CODE REVIEW

File: packages/server/db/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const { ethers } = require("ethers");
+const path = require("path");
+
+const dbPath = path.join(__dirname, "tokens");
+
+const tokens = levelup(leveldown(dbPath));
+
+const getRandomTokenId = () => ethers.BigNumber.from(ethers.utils.randomBytes(32)).toString();
+
+exports.getOrStoreRandomId = async (url) => {
```
Comment (angelagilhotra) [inline packages/server/db/index.js:12]: I'm just using a random uint256 generator and a simple key,value store to keep track of token ids; open to suggestions/ideas/feedback here!

**Comment by andytudhope** [comment]
So, we don't need a db at all and I don't want to add one.

The tokenId can be anything, is stored already in the contract, _and_ is returned by our subgraph. There is no need to stick it in a db here too.

The intention with this server is that we ought to be able to pick it up and use it anywhere without any kind of migration, except for making sure that the signer and arweave key move with it. That rules out a db.

[As I said here](https://kernel-community.slack.com/archives/C031ES44SA3/p1652884941887959?thread_ts=1652864472.892929&cid=C031ES44SA3), I think it is best if we just use a hash of the selected text (+ maybe the timestamp to ensure unicity). 

Please don't add a db to an application which rages against such machines :wink: :pray:

**Comment by andytudhope** [comment]
Reopened with massive amounts of humble pie as we are not able to reliably limit requests to the api to a specific domain (too easily spoofed) and so we need a better way to check the validity of a given request.

The only way we can think of atm is to check that the metadataUrl passed in is one that the server has previously created, otherwise return an error.

My sincerest apologies to dearest Angela :pray:

**Comment by angelagilhotra** [comment]
continuing on https://github.com/kernel-community/signature-economies/pull/56

**Pull request closed without merging**